### PR TITLE
Settings improvements

### PR DESCRIPTION
**Pull request by peppy**
- No longer allow sidebar to overlap actual content
- Close sidebar on click
- Fix misalignment when scrolling to a section
- Adjust vertical margin on an ongoing basis based on toolbar visibility

### CODE REVIEW

File: osu.Game/Overlays/SettingsOverlay.cs [hunk 1]
```diff
@@ -93,7 +96,11 @@ private void load(OsuGame game)
                         new SidebarButton
                         {
                             Section = section,
-                            Action = b => sectionsContainer.ScrollContainer.ScrollTo(b),
+                            Action = s =>
+                            {
+                                sectionsContainer.ScrollTo(s);
+                                sidebar.State = ExpandedState.Contracted;
```
Comment (smoogipooo) [inline osu.Game/Overlays/SettingsOverlay.cs]: This feels super weird... You move onto an icon to click it, suppose you wait around 750ms so it starts expanding, click, it contracts, then 750ms later it expands again.

Kinda preferred the feel before?

Comment (peppy) [inline osu.Game/Overlays/SettingsOverlay.cs]: why would you wait for 750ms?

Comment (peppy) [inline osu.Game/Overlays/SettingsOverlay.cs]: you're looking for a setting, the idea is you move off it within 750ms or you likely chose the wrong section and want to see the text again

Comment (smoogipooo) [inline osu.Game/Overlays/SettingsOverlay.cs]: I could be hovering down the list to get to another item than where my mouse is positioned - this is how I initially noticed this in the first place.

I don't want to be too stuck up on this but I just don't see the point given how settings acts in general (pushing the settings out, contracting them back in) is all.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**